Formula/qt@6.rb

```diff
@@ -0,0 +1,172 @@
+class QtAT6 < Formula
+  desc "Cross-platform application and UI framework"
+  homepage "https://www.qt.io/"
+  url "https://download.qt.io/official_releases/qt/6.0/6.0.1/single/qt-everywhere-src-6.0.1.tar.xz"
+  sha256 "d13cfac103cd80b216cd2f73d0211dd6b1a1de2516911c89ce9c5ed14d9631a8"
+  license all_of: ["GFDL-1.3-only", "GPL-2.0-only", "GPL-3.0-only", "LGPL-2.1-only", "LGPL-3.0-only"]
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on xcode: :build
+
+  depends_on "assimp"
+  depends_on "dbus"
+  depends_on "double-conversion"
+  depends_on "freetype"
+  depends_on "glib"
+  depends_on "icu4c"
+  depends_on "jpeg"
+  depends_on "libb2"
+  depends_on "libpng"
+  depends_on "libproxy"
+  depends_on "llvm"
+  depends_on "pcre2"
+  depends_on "python@3.9"
+  depends_on "zstd"
+
+  uses_from_macos "cups"
+  uses_from_macos "krb5"
+  uses_from_macos "perl"
+  uses_from_macos "sqlite"
+  uses_from_macos "zlib"
```

Comment:
https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/qt.rb#L28-L35 this is the QT 5 dependency list. Did QT get so much more complex? Or are we building a bunch more?

Reply:
Qt source code include these dependencies, if not specify these components, qt will build these libraries for qt internal use

Reply:
Why are we suddenly building plugins with dependencies? Did upstream start doing that?

Reply:
There is a CI configuration list: https://wiki.qt.io/Qt_6.0_Tools_and_Versions#macos_10.14_x86_64_2